test/algebraic_simplification.cpp

```diff
@@ -596,3 +597,52 @@ TEST(algebraic_simplification, pass_property)
     ASSERT_TRUE(pass->get_property(pass::PassProperty::REQUIRE_STATIC_SHAPE));
     ASSERT_FALSE(pass->get_property(pass::PassProperty::CHANGE_DYNAMIC_STATE));
 }
+
+TEST(algebraic_simplification, transpose_static_shape)
+{
+    // add a simple graph with transpose
+    Shape shape_in{1, 2};
+    auto param = make_shared<op::Parameter>(element::boolean, shape_in);
+    auto constant_perm = make_shared<op::Constant>(element::i64, Shape{2}, vector<int64_t>{1, 0});
+    auto transpose = make_shared<op::Transpose>(param, constant_perm);
+    auto f = make_shared<Function>(transpose, ParameterVector{param});
+
+    pass::Manager pass_manager;
+    pass_manager.register_pass<pass::AlgebraicSimplification>();
+    pass_manager.run_passes(f);
+
+    ASSERT_EQ(count_ops_of_type<op::Transpose>(f), 0);
+}
+
+TEST(algebraic_simplification, transpose_partial_shape)
+{
+    // add a simple graph with transpose
+    auto ps = PartialShape{1, Dimension::dynamic()};
+    auto param = make_shared<op::Parameter>(element::boolean, ps);
+    auto constant_perm = make_shared<op::Constant>(element::i64, Shape{2}, vector<int64_t>{1, 0});
+    auto transpose = make_shared<op::Transpose>(param, constant_perm);
+    auto f = make_shared<Function>(transpose, ParameterVector{param});
```

Comment:
@diyessi I added the change in algebraic_simplification pass. And this test with partial shape fails now, because `f->is_dynamic() == true` and `algebraic_simplification` pass only works on static functions with `set_property(PassProperty::REQUIRE_STATIC_SHAPE, true);`. 
The test could pass with setting `algebraic_simplification` pass `PassProperty::CHANGE_DYNAMIC_STATE`. Is this ok or we want to keep it for static shape only?

Reply:
Probably nothing in the pass tests that shapes are static before doing things to the ops.